src/python/pants/engine/fs_test.py

```diff
@@ -663,7 +663,7 @@ def test_snapshot_subset_globs_2(self) -> None:
 
         subset_snapshot = self.request_single_product(Snapshot, ss)
         assert set(subset_snapshot.files) == {"a.txt", "c.txt", "subdir2/a.txt"}
-        assert set(subset_snapshot.dirs) == {"subdir2/nested_subdir"}
```

Comment:
It feels like this should contain `{"subdir2/nested_subdir", "subdir2"}`

If we were matching this glob against the filesystem, `subdir2/nested_subdir` would be included because `subdir2/*` matches it. Its _content_ might not be, but it would.

Reply:
Oh wow, yeah, that's totally right. I will make that work.

Reply:
Done!